---
title: "Download and Begin Processing Data"
author: "Alexander Ham"
date: "`r Sys.Date()`"
output: html_document
---

# 1.

First I downloaded the tsv file from the project's ENA page containing sample accessions, run accessions, sample titles, and FASTQ links. I then used egrep to only get lines corresponding to beta cells (since alpha cells are also present in this dataset). "beta" is contained in the sample titles of beta cell samples.

```{}
wget -O fastq_link.tsv "ftp://www.ebi.ac.uk/ena/portal/api/filereport?accession=PRJEB30761&res
ult=read_run&fields=run_accession,fastq_ftp,sample_title&format=tsv&download=true&limit=0"

egrep "beta" fastq_link.tsv > beta_links.tsv
```

Then I wrote a script to download FASTQ files for all beta cell samples. Each run has two FASTQ files, corresponding to each paired end read. I wrote a script that downloads each files and names it with the run accession number, cell type, condition, and paired-end number. 

```{}
#!/bin/bash
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=get_beta_fastq
#SBATCH --time=02:00:00
#SBATCH --mem=4G

while read line; do
        run=$(echo "$line" | cut -f2)
        link=$(echo "$line" | cut -f3)
        link1="ftp://"$(echo "$link" | cut -d';' -f1)
        link2="ftp://"$(echo "$link" | cut -d';' -f2)
        condition=$(echo ${line} | egrep "Beta_...." -o)

        filename=${run}"_"${condition}
        echo ${link1} | xargs wget -O ${filename}_1.fastq.gz
        echo ${link2} | xargs wget -O ${filename}_2.fastq.gz


done < beta_links.tsv

exit
```

I got the files from the following project on ENA:
https://www.ebi.ac.uk/ena/browser/view/PRJEB30761

The data is from the 2019 publication
**High-fat diet impacts more changes in beta-cell compared to alpha-cell transcriptome**
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407777/

The data was generated by the authors at University of Geneva Medical School.

RNA was extracted using poly-A selection. 

Libraries were generated using Illumina TruSeq v2, which is an unstranded protocol. 

Beta cell and alpha cells were sequenced in this study. However, I am only looking at samples containing beta cells. 

The experimental condition is the diet of the mice, high-fat diet (HFD) or low-fat diet (LFD).

Illumina Hiseq2000 is the sequencing platform used. 

# 2.

I downloaded the Refseq GRCm39 mouse reference genome and GTF file: https://www.ncbi.nlm.nih.gov/assembly/GCF_000001635.27

```{}
wget -O GRCm39_genome.fa.gz 'fttp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_genomic.fna.gz'

wget -O GRCm39.gtf.gz "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27
_GRCm39/GCF_000001635.27_GRCm39_genomic.gtf.gz"
```


I gunzipped both files
```{}
gunzip GRCm39_genome.fa.gz

gunzip GRCm39.gtf.gz
```


Then I used STAR aligner to index the genome.   
```{}
#!/bin/bash -l

#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=index_genome
#SBATCH --time=05:00:00
#SBATCH --mem=60G

mamba activate angsd

STAR --runMode genomeGenerate \
--runThreadN 1 \
--genomeDir GRCm39_STARindex \
--genomeFastaFiles GRCm39_genome.fa \
--sjdbGTFfile GRCm39.gtf \
--sjdbOverhang 99

exit

```


Then I aligned one of the FASTQ files to the indexed genome. Using the Estimating Intron Sizes instructions on canvas, I found that the mouse genome assembly I am using has a minimum intron length of 1 and a maximum length of 1067346. I set the alignIntronMin parameter to 1 and the alignIntronMax 1068000.

```{}
#!/bin/bash -l

#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=align_reads
#SBATCH --time=02:00:00
#SBATCH --mem=36G

mamba activate angsd

STAR --runMode alignReads \
--runThreadN 2 \
--genomeDir GRCm39_STARindex \
--readFilesIn ERR3094492_Beta_LFD1_1.fastq.gz ERR3094492_Beta_LFD1_2.fastq.gz \
--readFilesCommand zcat \
--outFileNamePrefix ./ \
--outSAMtype BAM SortedByCoordinate \
--alignIntronMin 1 \
--alignIntronMax 1068000

exit

```


Then I ran FastQC on each of the FASTQ files I just aligned using STAR. I also ran MultiQC on the log file of the alignment. 

```{}
#!/bin/bash -l

#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=run_qc
#SBATCH --time=02:00:00
#SBATCH --mem=8G

mamba activate angsd

fastqc ERR3094492_Beta_LFD1_1.fastq.gz --extract
fastqc ERR3094492_Beta_LFD1_2.fastq.gz --extract

mamba activate multiqc

multiqc .

exit
```

The result of the FastQC indicates that there are issues with FASTQ files that I used for alignment. First of all, the Per base sequence content shows bad results for around the first 10 bp, with certain bases being overrepresented at each position. 

![](C:\Users\hamal\OneDrive\Desktop\per_base_content.jpg)

Additionally, there are significantly large Sequence Duplication Levels and Overrepresented Sequences in each.

![](C:\Users\hamal\OneDrive\Desktop\seq_duplication.jpg)

These factors indicate to me that there is probably adapter contamination in these samples. The Adapter Content did not show anything in the report, but FastQC can only detect a small subset of adapters. Trimming the FASTQ files could be a way to improve the read qualities. 

The MultiQC showed that 69.6% of reads were unmapped due to being too short. This could be because of short adapter sequences contaminating the reads. 

![](C:\Users\hamal\OneDrive\Desktop\alignment scores.jpg)

27.5% of reads were uniquely mapped and 2.8% of reads were mapped to multiple loci, so out of the reads that were not too short, a large proportion were properly mapped to one locus. 